### PR TITLE
Add MusicBrainz Provider

### PR DESCRIPTION
**Pull request by andrzejkupczyk**
[MusicBrainz](https://musicbrainz.org/) is an open music encyclopedia that collects music metadata and makes it available to the public.

https://musicbrainz.org/doc/Development/OAuth2

### CODE REVIEW

File: src/MusicBrainz/MusicBrainzExtendSocialite.php [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+namespace SocialiteProviders\MusicBrainz;
+
+use SocialiteProviders\Manager\SocialiteWasCalled;
+
+class MusicBrainzExtendSocialite
+{
+    /**
+     * Register the provider.
+     */
```
Comment (lucasmichot) [inline src/MusicBrainz/MusicBrainzExtendSocialite.php:11]: Missing `@param` declaration here

File: src/MusicBrainz/Provider.php [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+use SocialiteProviders\Manager\OAuth2\User;
+
+class Provider extends AbstractProvider
+{
+    /**
+     * Unique Provider Identifier.
+     */
+    public const IDENTIFIER = 'MUSICBRAINZ';
+
+    protected $scopes = ['profile'];
```
Comment (lucasmichot) [inline src/MusicBrainz/Provider.php:16]: Missing `@inheritdoc` declaration here

File: src/MusicBrainz/Provider.php [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+    protected $scopeSeparator = ' ';
```
Comment (lucasmichot) [inline src/MusicBrainz/Provider.php:18]: same here

File: src/MusicBrainz/Provider.php [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+    protected $usesPKCE = true;
```
Comment (lucasmichot) [inline src/MusicBrainz/Provider.php:20]: same here

File: src/MusicBrainz/Provider.php [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        return (new User())->setRaw($user)->map([
+            'id'   => $user['metabrainz_user_id'],
+            'name' => $user['sub'],
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getTokenFields($code)
```
Comment (lucasmichot) [inline src/MusicBrainz/Provider.php:72]: This method can be removed

File: src/MusicBrainz/README.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+Configure the package's listener to listen for `SocialiteWasCalled` events.
+
+Add the event to your `listen[]` array in `app/Providers/EventServiceProvider`. See the [Base Installation Guide](https://socialiteproviders.com/usage/) for detailed instructions.
+
+```php
+protected $listen = [
+    \SocialiteProviders\Manager\SocialiteWasCalled::class => [
+        // ... other providers
+        'SocialiteProviders\\MusicBrainz\\MusicBrainzExtendSocialite@handle',
```
Comment (lucasmichot) [inline src/MusicBrainz/README.md:31]:
```suggestion
        \SocialiteProviders\MusicBrainz\MusicBrainzExtendSocialite::class.'@handle',
```

**Review by lucasmichot — CHANGES_REQUESTED** [review]
Thanks for the PR @andrzejkupczyk 
Can you adjust the PR with the review comments ?

**Review by lucasmichot — APPROVED** [review]
Thanks for this PR @andrzejkupczyk 
GTM @atymic 👍

**Comment by andrzejkupczyk** [comment]
Thanks for the review, @lucasmichot 👍🏻

**Review by atymic — APPROVED** [review]

**Pull request merged**